pkg/route/api/helper.go

```diff
@@ -0,0 +1,16 @@
+package api
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+// IngressConditionStatus returns the first status and condition matching the provided ingress condition type.
```

Comment:
It may be that I don't understand conditions but can/should a later condition invalidate an earlier one?